### ISSUE TITLE
False Positive and Incorrect Error Message for Style/AccessModifierDeclarations

### ISSUE THREAD

**Issue by Mihail-K** [rubocop-hq/rubocop#7740]
RuboCop 0.80.0 is reporting offences for marking existing methods as private, even when the method is not defined within the scope it is marked private in.
Furthermore, the message produced by the offence is misleading.

--------

## Expected behavior
I expect to be able to mark methods introduced by reflection or otherwise defined outside of the current scope as protected or private.

## Actual behavior
Consider a hypothetical example using a Rails model:
```ruby
ApplicationRecord.connection.create_table(:articles) do |t|
  t.string :name
  t.string :slug
end

class Article < ApplicationRecord
  def name=(value)
    super
    self.slug = name&.parameterize
  end

  private :slug=
end
```

The rubocop offence is reported on the `private :slug=` line:
> `private` should not be inlined in method definitions. (convention:Style/AccessModifierDeclarations)

The reported offence is incorrect since:
* The line is not a method definition, hence the error message is incorrect and misleading.
* The line affects a method that is not defined by application code. It is created by Rails based on the connected database schema, and its natural access modifier is outside of application control.

## Steps to reproduce the problem
Minimum case to reproduce:
```ruby
module Bar
  def bar
    nil
  end
end

class Foo
  include Bar
  private :bar
end
```
The error message is produced:
> `private` should not be inlined in method definitions. (convention:Style/AccessModifierDeclarations)

The line neither defines a method, nor does class `Foo` itself declare the method `bar`.

## RuboCop version
```
0.80.0 (using Parser 2.7.0.2, running on ruby 2.7.0 x86_64-linux)
```

**Comment by Drenmi** [comment]
I think that we can skip registering an offense when the argument is a symbol literal, e.g.:

```ruby
private :foo
```
